massa-grpc/src/tests/public.rs

```diff
@@ -0,0 +1,1707 @@
+        max_function_name_length: MAX_FUNCTION_NAME_LENGTH,
+        max_parameter_size: MAX_PARAMETERS_SIZE,
+        max_operations_per_message: MAX_OPERATIONS_PER_MESSAGE,
+        genesis_timestamp: *GENESIS_TIMESTAMP,
+        t0: T0,
+        periods_per_cycle: PERIODS_PER_CYCLE,
+        keypair: keypair.clone(),
+        max_channel_size: 128,
+        draw_lookahead_period_count: 10,
+        last_start_period: 0,
+        max_denunciations_per_block_header: MAX_DENUNCIATIONS_PER_BLOCK_HEADER,
+        max_addresses_per_request: 50,
+        max_slot_ranges_per_request: 50,
+        max_block_ids_per_request: 50,
+        max_endorsement_ids_per_request: 100,
+        max_operation_ids_per_request: 250,
+        max_filters_per_request: 32,
+        server_certificate_path: PathBuf::default(),
+        server_private_key_path: PathBuf::default(),
+        certificate_authority_root_path: PathBuf::default(),
+        client_certificate_authority_root_path: PathBuf::default(),
+        client_certificate_path: PathBuf::default(),
+        client_private_key_path: PathBuf::default(),
+        max_query_items_per_request: 50,
+    };
+
+    let mip_stats_config = MipStatsConfig {
+        block_count_considered: MIP_STORE_STATS_BLOCK_CONSIDERED,
+        warn_announced_version_ratio: Ratio::new_raw(30, 100),
+    };
+
+    let mip_store = MipStore::try_from(([], mip_stats_config)).unwrap();
+
+    let service = MassaPublicGrpc {
+        consensus_controller: Box::new(consensus_controller),
+        consensus_channels,
+        execution_controller: execution_ctrl.0.clone(),
+        execution_channels: ExecutionChannels {
+            slot_execution_output_sender,
+        },
+        pool_channels: PoolChannels {
+            endorsement_sender,
+            operation_sender,
+            selector: selector_ctrl.0.clone(),
+            execution_controller: execution_ctrl.0.clone(),
+        },
+        pool_controller: pool_ctrl.0,
+        protocol_controller: Box::new(MockProtocolController::new()),
+        protocol_config: ProtocolConfig::default(),
+        selector_controller: selector_ctrl.0,
+        storage: shared_storage,
+        grpc_config: grpc_config.clone(),
+        version: *VERSION,
+        node_id: NodeId::new(keypair.get_public_key()),
+        keypair_factory: KeyPairFactory {
+            mip_store: mip_store.clone(),
+        },
+    };
+
+    let stop_handle = service.serve(&grpc_config).await.unwrap();
+    // std::thread::sleep(Duration::from_millis(100));
+
+    // start grpc client and connect to the server
+    let channel = tonic::transport::Channel::from_static("grpc://localhost:8888")
+        .connect()
+        .await
+        .unwrap();
+
+    let mut res = PublicServiceClient::new(channel);
+
+    let s = res.get_status(GetStatusRequest {}).await.unwrap();
+    dbg!(s);
+    stop_handle.stop();
+    std::thread::sleep(Duration::from_millis(1000));
+}
+
+#[tokio::test]
+#[serial]
```

Comment:
Remove serial could boost our test execution, we should find a workaround

Reply:
done